compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partition_cut.rs

```diff
@@ -9,39 +9,126 @@ use crate::optimization::dag::multi_parameters::partitions::PartitionIndex;
 use crate::optimization::dag::solo_key::analyze::out_variances;
 use crate::optimization::dag::solo_key::symbolic_variance::SymbolicVariance;
 
+use super::optimize::MacroParameters;
+
 const ROUND_INNER_MULTI_PARAMETER: bool = false;
 const ROUND_EXTERNAL_MULTI_PARAMETER: bool = !ROUND_INNER_MULTI_PARAMETER && true;
 
+#[derive(Clone, Debug)]
+pub struct ExternalPartition {
+    pub name: String,
+    pub macro_params: MacroParameters,
+    pub max_variance: f64,
```

Comment:
maybe a comment saying it's the maximimum allowed variance for any ciphertext in that partition in absolute scale